app/file-detail/controller.js

```diff
@@ -1,25 +1,108 @@
+    }),
+
+    linkedInUrl: computed('model.file', 'fileUrl', function() {
+        return `https://www.linkedin.com/cws/share?url=${this.get('fileUrl')}&title=${this.get('model.file.name')}`;
+    }),
+
+    emailUrl: computed('model.file', 'fileUrl', function() {
+        return `mailto:?subject=${this.get('model.file.name')}&body=${this.get('fileUrl')}`;
+    }),
+
+    mfrUrl: computed('model.file', function() {
+        return `https://mfr.osf.io/render?url=${window.location.href}?action=download%26mode=render`;
+    }),
+
+    shareiFrameDynamic: computed('model.file', function() {
+        return `<style>.embed-responsive{position:relative;height:100%;}.embed-responsive iframe{position:absolute;height:100%;}</style><script>window.jQuery || document.write('<script src="//code.jquery.com/jquery-1.11.2.min.js">\x3C/script>') </script><link href="https://mfr.osf.io/static/css/mfr.css" media="all" rel="stylesheet"><div id="mfrIframe" class="mfr mfr-file"></div><script src="https://mfr.osf.io/static/js/mfr.js"></script> <script>var mfrRender = new mfr.Render("mfrIframe", "${this.get('mfrUrl')}");</script>`;
+    }),
+
+    shareiFrameDirect: computed('model.file', function() {
+        return `<iframe src="${this.get('mfrUrl')}" width="100%" scrolling="yes" height="677px" marginheight="0" frameborder="0" allowfullscreen webkitallowfullscreen>`;
+    }),
+
+    fileTags: computed('model.file', function() {
+        return this.get('model.file.tags');
     }),
+
+    fileVersions: computed('model.file', function() {
+        return $.getJSON(`${this.get('model.file.links.download')}?revisions=&`).then(this._returnFileVersion.bind(this));
+    }),
+
+    edit: computed('currentUser', 'model.user', function() {
+        let _edit = false;
+        if (this.get('model.user.id')) {
+            _edit = (this.get('model.user.id') === this.get('currentUser.currentUserId'));
+        }
+        return _edit;
+    }),
+
     actions: {
-        download() {
-            window.location = this.get('model.links.download');
+        share() {
+            document.querySelector('#sharePaneUrl').select();
+            document.execCommand('copy');
+        },
+
+        download(version) {
+            const url = `${this.get('model.file.links.download')}?revision=${version}`;
+            window.location = url;
+        },
+
+        changeView() {
+            $('#mfrIframeParent').toggle();
+            $('#revisionsPanel').toggle();
+            $('.view-button').toggleClass('btn-default btn-primary');
         },
+
         openFile(file) {
-            if (file.get('guid')) {
-                this.transitionToRoute('file-detail', file.get('guid'));
-            } else {
-                window.location = `/file_redirect${file.get('path')}`;
-            }
+            const fileID = file.get('guid') ? file.get('guid') : file.id;
+            this.transitionToRoute('file-detail', fileID);
```

Comment:
We need to be using the logic you removed here. Just using file-id won't work, we need to go to file-redirect using the path